src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -1113,6 +1116,79 @@ bool isLambdaConversion(BoundExpression expr)
+            // clear the execution stack before allocation. (See corresponding requirement
+            // for 'stackalloc' in LocalRewriter.VisitStackAllocArrayCreationBase().)
+
+            // For now, simply allocate the array on the heap: new Span<T>(new T[length] { ... })
+            var array = CreateParamArrayArgument(
+                syntax,
+                ArrayTypeSymbol.CreateSZArray(compilation.SourceAssembly, TypeWithAnnotations.Create(elementType)),
+                arrayArgs,
+                compilation,
+                diagnostics);
+
+            var spanConstructor = (MethodSymbol)Binder.GetWellKnownTypeMember(compilation, WellKnownMember.System_Span_T__ctorArray, diagnostics, syntax.Location);
+            if (spanConstructor is null)
+            {
+                return createBadExpression(syntax, paramArrayType);
+            }
+
+            var spanType = spanConstructor.ContainingType.Construct(elementType);
+            spanConstructor = spanConstructor.AsMember(spanType);
+
+            BoundExpression span = new BoundObjectCreationExpression(
+                syntax,
+                spanConstructor,
+                ImmutableArray.Create(array))
+            { WasCompilerGenerated = true };
+
+            if (paramArrayType.IsReadOnlySpanType(compilation))
+            {
+                var spanToReadOnlySpanOperator = (MethodSymbol)Binder.GetWellKnownTypeMember(compilation, WellKnownMember.System_Span_T__op_Implicit_SpanReadOnlySpan, diagnostics, syntax.Location);
+                if (spanToReadOnlySpanOperator is null)
+                {
+                    return createBadExpression(syntax, paramArrayType);
+                }
+
+                spanToReadOnlySpanOperator = spanToReadOnlySpanOperator.AsMember(spanType);
+
+                // Convert Span<T> to ReadOnlySpan<T>.
+                span = new BoundCall(
+                    syntax,
+                    receiverOpt: null,
+                    method: spanToReadOnlySpanOperator,
+                    arguments: ImmutableArray.Create<BoundExpression>(span),
+                    argumentNamesOpt: default,
+                    argumentRefKindsOpt: default,
+                    isDelegateCall: false,
+                    expanded: false,
+                    invokedAsExtensionMethod: false,
+                    argsToParamsOpt: default,
+                    defaultArguments: default,
+                    resultKind: LookupResultKind.Viable,
+                    type: spanToReadOnlySpanOperator.ReturnType)
+                { WasCompilerGenerated = true };
+            }
+
+            return span;
+
+            static BoundExpression createBadExpression(SyntaxNode syntax, TypeSymbol type)
+            {
+                return new BoundBadExpression(syntax, LookupResultKind.NotReferencable, ImmutableArray<Symbol?>.Empty, ImmutableArray<BoundExpression>.Empty, type)
```

Comment:
>ImmutableArray<BoundExpression>.Empty [](http://example.com/codeflow?start=119&length=37)

```arrayArgs```? Even in a missing helper scenarios we want IOperation tree to include values from source.

Reply:
A PROTOTYPE comment will be fine for now, I think.